### DISCUSSION TITLE
Step into the community that has formed around 1brc

### DISCUSSION THREAD

**Discussion by jcbhmr** [gunnarmorling/1brc#275] [General]
Hello @gunnarmorling ! 👋

I really like how much **cool code** I've come across while reading the code from the Java submissions and the other Show and Tell submissions in other languages. It's been a really educational experience!

I think it would  be a good idea to move this challenge to a  GitHub organization so that other community  members can more "officially"-ish manage their own parallel 1brc-cpp, 1brc.js, 1brc-zig, 1brc.rs etc. challenges. This would probably entail deputizing other members of the community that have shown that they are interested enough in the 1brc idea to make/share other language solutions

To that end, I think a https://github.com/1brc organization is a good idea. I made some initial efforts towards a language-agnostic @1brc organization and a https://1brc.dev/ website. I am particularily interested in formalizing the submissions from other non-Java languages. I think it would be a good idea to get other community members involved in the process so that you aren't the sole approver/tester person - especially if other non-Java submissions get formalized.

**I do not expect the core Java-original prize or anything to change.** The Java-centric focus is great! I just think it would be good for this 1brc project to grow a bit into the community that formed around it.

To that end, I think @candrewlee14 @tumdum @Edgar-P-yan @AlexanderYastrebov @ifnesi @dannyvankooten or others could get involved if they so wish (they were interested enough to create a separate language solution lol) to provide other-language hubs for submissions. If you don't want to  that's perfectly OK ❤️

This is me trying to provide a half-baked solution up-front so that it's easy to run with it if you like it 👍 don't worry though you can always say no and I'm OK with that too 😊

**Comment by Edgar-P-yan** [comment]
@jcbhmr Hi! Sounds interesting to me) I am down to post my nodejs/bun solution into the @1brc organization, also the website you made looks cool

**Reply by jcbhmr** [reply]
Cool! I invited you when I made the post (so three days ago 😅 lol). What I'm hoping for is a centralized @1brc organization to host all the different languages (probably each in their own repo). I'm trying to spearhead stuff here and open this awesome idea for a competition up to more languages! 🤩

So to the "what I want from you" is basically:
1. **Are you willing to "manage" the JavaScript submissions?** Do you want to hand that off to someone else from https://github.com/gunnarmorling/1brc/discussions/categories/show-and-tell ? <sup>or i guess ill do it if no one else will</sup>
2. Do you want to manually test things AND/OR use something like https://buildjet.com/for-github-actions or https://www.warpbuild.com/ to get a beefy GitHub Actions runner (both offer free trials iirc)?
3. Should it be under @1brc or should https://1brc.dev just point to your https://github.com/Edgar-P-yan/1brc-nodejs-bun repository for submissions?

**Reply by Edgar-P-yan** [reply]
@jcbhmr Hi! Sorry for the delay.
1. Yes
2. I don't know yet, i don't think there will be a lot of submissions at the begining so i am cool with manually testing on my end.
3. Under @1brc would be more elegant. It seems like the invitation to the org has been expired, could you please send me a new one? This weekend i'll clean things up and push the repo to the organization.

**Reply by jcbhmr** [reply]
Resent 👍 and sounds good!

> I don't know yet, i don't think there will be a lot of submissions at the begging so i am cool with manually testing on my end.

I agree 👍

**Reply by Edgar-P-yan** [reply]
@jcbhmr Hi! Yesterday i pushed two repositories https://github.com/1brc/nodejs for nodejs and https://github.com/1brc/bun for bun. I though having them separate is more fair as those runtimes offer very different APIs, optimisation techniques and etc, and honestly bun outperforms node.js a lot in this specific task so having them compared in the same leaderboard would be unfair.

Here is what's different from the original repo:
- removed all the Java programs and Java-based solutions, except the ones that generate the `measurements.txt` file, and those that do the testing;
- aside from the optimised solution that i implemented earlier i also created a baseline solution without any optimisations as a simple reference;
- updated the README.md to reflect the node.js/bun specifics, updated the leaderboard;

I think those repos could be used as a template for future repositories dedicated to other languages

**Comment by krk** [comment]
Maybe we could find sponsorship to run all the benchmarks on the same machine as the Java one, Hetzner CCX33?

**Reply by tarsa** [reply]
> the same machine as the Java one, Hetzner CCX33?

it changed. see discussion at: https://github.com/gunnarmorling/1brc/issues/189

now it's: https://github.com/gunnarmorling/1brc?tab=readme-ov-file#results
> These are the results from running all entries into the challenge on eight cores of a [Hetzner AX161](https://www.hetzner.com/dedicated-rootserver/ax161) dedicated server (32 core AMD EPYC™ 7502P (Zen2), 128 GB RAM).

**Comment by gunnarmorling** [comment]
Hey guys, sorry for the late reply. Just wanted to let you know that I'm aware of this thread and will hopefully reply with a more meaningful reply in a bit. Right now, I'm focussed on getting the existing 1BRC challenge as-is over the finishing line on Jan 31. I think it could then be a good time to explore this further.

**Reply by rofrol** [reply]
I have found this for zig:
- https://github.com/wexder/1brc-zig
  - https://neurobug.com/posts/zig/billion/
  - https://www.reddit.com/r/Zig/comments/1bkvkhn/one_billion_rows_challenge_in_zig/
- https://github.com/candrewlee14/1brc-zig
  - https://github.com/gunnarmorling/1brc/discussions/180
- https://github.com/GoWind/1brc
  - https://zig.news/gowind/from-500-secs-to-35-the-1brc-challenge-5b22
- https://github.com/almmiko/1brc-zig
  - https://github.com/gunnarmorling/1brc/discussions/569
  - https://www.linkedin.com/posts/alexander-mykolaichuk_1brc-zig-activity-7158415294309298176-BdbN/
